src/helpers/xml-sitemap-helper.php

```diff
@@ -74,66 +77,36 @@ public function find_images_for_indexables( $indexables, $type = 'image-in' ) {
 			];
 		}
 
-		// The featured image may not be in the links table. Make sure it is included in the image list.
-		foreach ( $indexable_ids as $object_id => $indexable_id ) {
-			if ( ! is_int( $object_id ) || $object_id <= 0 ) {
-				continue;
-			}
-
-			$featured_image_link = $this->get_featured_image_link( $object_id );
-
-			if ( is_null( $featured_image_link ) ) {
-				continue;
-			}
-
+		$placeholders    = \implode( ', ', \array_fill( 0, count( $indexable_ids ), '%s' ) );
+		$indexable_table = Model::get_table_name( 'Indexable' );
+		// phpcs:disable WordPress.DB.PreparedSQLPlaceholders.UnfinishedPrepare -- Placeholders are in $placeholders.
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Table names can not be prepared.
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery -- Query can not be done without a direct query.
+		$featured_images = $wpdb->get_results(
+			$wpdb->prepare(
+				"
+				SELECT pm.post_id, i.permalink
+				FROM {$wpdb->prefix}postmeta AS pm
+				INNER JOIN $indexable_table AS i ON i.object_id = pm.meta_value AND i.object_type = 'post'
```

Comment:
I think we can do without the join, we have post_ids in `array_keys($indexable_ids);`

Reply:
I want to retrieve a list of the featured images, so the post_ids are used to get all postmeta rows with `_thumbnail_id` when then I join the meta_value, the featured image ID, on our indexables to get the permalink.